Proposal: Randy Agudelo, René Borner, Caitlin Enright, Elie Hess

eliehess opened the pull request:
GitHub usernames: RandyAgudelo, retat, eliehess, cenright71925

eliehess committed:
Create Proposal.md

eliehess committed:
Update Proposal.md

charlieroberts commented:
This sounds good to me! Note that there's no way to get Spotify audio into the web audio API, so you'll have to use data that the API gives you to drive your visualization.

RandyAgudelo committed:
Server (#2)

retat committed:
removed view engine

Commit:
removed view engine (#3)

retat committed:
Merge remote-tracking branch 'origin/master'

retat committed:
added login and registration functionality

retat committed:
added favicon

cenright71925 committed:
Merge pull request #4 from cs4241-best-group/login-and-registration

Login and registration

cenright71925 committed:
Made preliminary layout for home page

RandyAgudelo committed:
Spotify Integration

cenright71925 committed:
Added caption to sidebar

retat committed:
Merge remote-tracking branch 'origin/homepage-layout' into Spotify-Integration

RandyAgudelo committed:
Access token added

retat committed:
added spotify player with play and pause functionality

RandyAgudelo committed:
Merge remote-tracking branch 'origin/include-spotify-player' into Spotify-Integration

retat committed:
Merge pull request #5 from cs4241-best-group/homepage-layout

Homepage layout

retat committed:
Merge branch 'include-spotify-player' into Spotify-Integration

eliehess committed:
Minor changes

Delete dropdown, clean up login page, fix typos, code style

retat committed:
working on spotify token connection to backend

RandyAgudelo committed:
Merge remote-tracking branch 'origin/Spotify-Integration' into Spotify-Integration

eliehess committed:
Undo some unnecessary changes

retat committed:
Merge pull request #6 from cs4241-best-group/cleanup

Cleanup

eliehess committed:
Merge branch 'master' into username

eliehess committed:
Merge branch 'master' into username

# Conflicts:
#	public/home.html

retat committed:
added streaming scope so token works!

RandyAgudelo committed:
Merge remote-tracking branch 'origin/Spotify-Integration' into Spotify-Integration

eliehess committed:
Add get current user

Changes:
- add ability to get "/user" from app.js, which returns the currently-logged-in user
- change default port to 3001 instead of 3000 because 3000 is already occupied for some reason
- apply default WebStorm code styling
- navbar now updates with the currently-logged-in username

eliehess committed:
Merge branch 'Spotify-Integration' into username

# Conflicts:
#	app.js

eliehess committed:
Logout actually works :shockedpikachuface:

retat committed:
Merge pull request #7 from cs4241-best-group/username

Username

cenright71925 committed:
Added bootsrap table with sample data

cenright71925 committed:
Push to spotify branch

eliehess committed:
Add default image if song not found

retat committed:
backend route to create recommendations

retat committed:
Merge remote-tracking branch 'origin/save-recommendations-to-db' into homepage-layout

# Conflicts:
#	public/home.html

eliehess committed:
Merge branch 'songnotfound' into homepage-layout

# Conflicts:
#	public/home.html
#	public/stylesheets/style.css

retat committed:
fixed play and pause button

cenright71925 committed:
Added scrollable song list

cenright71925 committed:
resolved merge conflicts

cenright71925 committed:
Added hidden ID field to the table

retat committed:
search works with correct spotify results and shows them in a table

cenright71925 committed:
Added hidden Id field to table with real results

eliehess committed:
Add get("/recommendation")

eliehess committed:
Fix incorrect variable usage

eliehess committed:
Autoformat!

eliehess committed:
Add autofocus to username field

retat committed:
added star rating

retat committed:
Merge remote-tracking branch 'origin/star-rating' into get-search-results-json

# Conflicts:
#	public/home.html

eliehess committed:
Non-functional (code executes out-of-order)

eliehess committed:
Get recommendations actually works

eliehess committed:
Simplify and delete commented-out code and console.log()s

eliehess committed:
Change variable names

eliehess committed:
Add recommendations from database to home page

eliehess committed:
Check number of entries before displaying them all

eliehess committed:
Add refresh button and error checking if there are no recommendations in the db

eliehess committed:
Mark comments to be deleted

eliehess committed:
Merge branch 'homepage-layout' into integration

eliehess committed:
Merge homepage-layout

eliehess committed:
Merge branch 'Spotify-Integration' into integration

eliehess committed:
Merge branch 'get-search-results-json' into integration

eliehess committed:
Merge

eliehess committed:
Fix bad things

eliehess committed:
Delete unnecessary table and revert port number

cenright71925 committed:
Fixed some container styling

cenright71925 committed:
Fixed formatting for side container

eliehess committed:
Merge branch 'get-search-results-json' into integration

RandyAgudelo committed:
Error Handling for non-premium account and identifying the webplayer to avoid errors

RandyAgudelo committed:
Merge remote-tracking branch 'origin/Spotify-Integration' into Spotify-Integration

# Conflicts:
#	app.js
#	public/js/createPlayer.js

RandyAgudelo committed:
attempt to push

eliehess committed:
Merge get-search-results

eliehess committed:
Bring back rateYo

eliehess committed:
Function prototype created

cenright71925 committed:
Fixed table styling (#8)

* Fixed table styling

* Comment out console.log()s

* Update deprecated function call

eliehess committed:
Merge branch 'integration' into add-recommendation

RandyAgudelo committed:
Merge remote-tracking branch 'origin/integration' into Spotify-Integration

# Conflicts:
#	app.js
#	public/js/createPlayer.js

eliehess committed:
Adding recommendations works

retat committed:
Merge pull request #9 from cs4241-best-group/Spotify-Integration

Merge remote-tracking branch 'origin/integration' into Spotify-Integration

eliehess committed:
Now submits track number and username

eliehess committed:
Fix warnings

retat committed:
Revert "Merge remote-tracking branch 'origin/integration' into Spotify-Integration" (#11)

eliehess committed:
Revert previous commit

eliehess committed:
Recommendations theoretically fully functional

retat committed:
it's possible to play other tracks now

eliehess committed:
Recommendations now show most recent first

eliehess committed:
Recommendations now use stars instead of numerals

eliehess committed:
Change stars to images instead of unicode

eliehess committed:
Delete old recommendation builder

cenright71925 committed:
Merge pull request #13 from cs4241-best-group/player-with-different-tracks

it's possible to play other tracks now

eliehess committed:
Merge branch 'integration' into add-recommendation

eliehess committed:
Remove unnecessary console.log

retat committed:
fix play pause

retat committed:
Merge pull request #14 from cs4241-best-group/add-recommendation

Add recommendation

retat committed:
Merge remote-tracking branch 'origin/integration' into hotfix-playpause

cenright71925 committed:
Merge pull request #15 from cs4241-best-group/hotfix-playpause

fix play pause

cenright71925 committed:
Added player header, worked on formatting

retat committed:
fixed wrong track id in recommendations

cenright71925 committed:
Merge pull request #16 from cs4241-best-group/recommendations-with-play-button

fixed wrong track id in recommendations

cenright71925 committed:
Merge formatting with working player and integration

retat committed:
Merge pull request #17 from cs4241-best-group/player-formatting

Player formatting

cenright71925 committed:
Centered the ratings on recomendations bar

cenright71925 committed:
Working on play pause button

retat committed:
fixed wrong track id in recommendations (#18)

cenright71925 committed:
Changed select song button to a play symbol

cenright71925 committed:
changed button colors to purple to fit the theme

cenright71925 committed:
changed button color to be pink

cenright71925 committed:
Updated color of log in buttons

cenright71925 committed:
Merge remote-tracking branch 'origin/integration' into ui-edits

# Conflicts:
#	public/home.html

retat committed:
Merge pull request #20 from cs4241-best-group/ui-edits

Ui edits

retat committed:
responsive layout (#19)

retat committed:
Background visual (#21)

* Testing for audio visualizer

* Tried to make it be a js file

* Placed audio Visualizer in the background

* merging

* saving current Track id in backend

* working on animation

* fixing play pause animation

* merging

retat committed:
layout fixes

cenright71925 committed:
Fixed play pause button issue when switching songs

cenright71925 committed:
clears the form when a recomendation is made

cenright71925 committed:
Merge remote-tracking branch 'origin/integration' into ui-edits

# Conflicts:
#	public/home.html
#	public/js/createPlayer.js

retat committed:
trying fixes

cenright71925 committed:
Changed the rating stars to purple

retat committed:
trying fixes

cenright71925 committed:
Made the rating card stars purple as well

cenright71925 committed:
deleted bad file

cenright71925 committed:
Merge pull request #23 from cs4241-best-group/turntable-fix-and-layout

Turntable fix and layout

cenright71925 committed:
Merge remote-tracking branch 'origin/integration' into ui-edits

# Conflicts:
#	public/home.html

cenright71925 committed:
Update merge conflicts

cenright71925 committed:
Update README.md

retat committed:
Merge pull request #25 from cs4241-best-group/cenright71925-readme

Update README.md

retat committed:
Merge remote-tracking branch 'origin/integration' into ui-edits

retat committed:
deleted unnecessary js src

retat committed:
Merge pull request #24 from cs4241-best-group/ui-edits

Ui edits

cenright71925 committed:
Merge pull request #26 from cs4241-best-group/integration

Update master branch

cenright71925 committed:
Integration (#37)

* Update README.md

* Update README.md

* Update README.md (#29)

* Update README.md (#30)

Ha ha I get to merge this PR without a review

* Delete unnecessary comments and autoformat

* Comment out all console.logs() so that they can be deleted

* Update README.md (#33)

* Delete Bug

* Updated the readME for my part (#34)

* Delete unnecessary things

* Delete get("/trackInfo")

Resolves #31

* Delete unnecessary console.log()s

* I missed a few things

* Fix scrolling (#36)

* added scrolling for small computers

* Switch port back to 3000

* Delete comments

* Update style.css

* Delete fullstar.png

* Fix?

* Update createPlayer.js

cenright71925 committed:
Update README.md (#38)

* Update README.md

* Update README.md

retat committed:
Update README.md

cenright71925 committed:
Merge pull request #39 from cs4241-best-group/retat-patch-1

Update README.md
